### PR TITLE
refactor: move to expo

### PR DESCRIPTION
**Pull request by Harjot1Singh**
## Summary

> Yes... please excuse the size of this 😅 

It's been a while since we've done significant work on this project. Since then, expo has become an excellent choice of tooling to reduce the overall maintenance burden. This PR fundamentally moves us over to expo, with a number of tooling changes that will simplify working with the project successfully.

### Expo CNG

Continuous native generation means that we no longer have to maintain specific `ios` and `android` folders - these are generated on build with the appropriate config. Much more stable & deterministic across different setups.

### Expo Router

Instead of setting up the screens and structure for defining screens ourselves, `expo-router` uses filesystem-based routing. Since the convention is public, it makes it much easier for everyone to know where and how to add new routes. It also comes with deeplinking OOTB. It's powered by `react-navigation` under-the-hood, which is what we were explicitly using beforehand.

### Atomic Design

Organising component code has been made clearer by adopting [Atomic Design](https://bradfrost.com/blog/post/atomic-web-design/). Components must fit into `atoms`, `molecules`, `organisms`, `templates`, and `templates` are always rendered by routes, to improve testability and reusability. We have not figured out a convention for layout routes just yet.

We should adopt this for our design library/system.

### Logging

We're using a proper logger, with module-scoped messages O_o

### Config

Env variables can be set in the `config/environment/config.{environment}.ts` files. Running the app with `APP_ENV=next` will inject the values in `config.next.ts` and so forth. TS files have been used to ensure we can catch missed declarations/fields.

By default, `APP_ENV=local`, so `config.localts` is used. This file is gitignored and so you should copy `config.example.ts` over as `config.local.ts`.

### Mise/Simplified Tooling

We now recommend using mise to install the different versions of tools, in `.ruby-version`, `.node-version`, and `.java-version`. You can also use other tools separately, such as `asdf`, `rvm`, and `nvm`.

### EAS build + submit

Now that we're using the `eas-cli` to build and submit (in local mode), there is no need for fastlane, homebrew (graphicsmagik), or ruby! Simplified tooling and reduced maintenance burden :)

### CODE REVIEW

File: Gemfile
Comment (saihaj) [inline Gemfile:1]: do we need gem file now? we can just use expo EAS to do builds and they have a really nice way to submit to App Store

File: src/with-contexts.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+          </GestureHandlerRootView>
+        </ThemeProvider>
+      </SafeAreaProvider>
+    </Provider>
+  )
+
+  return WithContexts
+}
+
+export default withContexts
```
Comment (saihaj) [inline src/with-contexts.tsx:44]: we should actually start to avoid default export, they make discoverability a nightmare

**Review by saihaj — DISMISSED** [review]
Thanks for reviving this back again!!!

File: Gemfile
Comment (Harjot1Singh) [reply Gemfile:1]: I did look into this. EAS builds are paid and we have bountiful GH action credits. 

We could run `eas build --local`, but I found that it runs `fastlane` under the hood. 

Maybe in the future, if the `eas` CLI can pretty much do everything for us.

File: src/with-contexts.tsx (hunk 1 above)
Comment (Harjot1Singh) [reply src/with-contexts.tsx:44]: So, I've heard a lot about this and looked into it a few times. I've not landed anywhere particularly strong on the subject.

I'm not convinced they actually do limit discoverability, unless you're using something like vim lol - VS Code and IntelliJ both handle default export usage lookups + refactors + renames quite well. 

A pro of using default exports (idiomatically) can be to encourage more thought around additional exports ala single-responsibility.

**Comment by Harjot1Singh** [comment]
Appreciate the review, @saihaj!

File: Gemfile
Comment (Harjot1Singh) [reply Gemfile:1]: Edit: for the life of me, I cannot get some of the fastlane stuff to work properly, and I've found expo build plugins for what we need. Going to give  it a shot.

Comment (bhajneet) [reply Gemfile:1]: Does this help? https://dev.to/rgomezp/how-to-set-up-an-eas-local-build-on-github-actions-1l0i

**Comment by bhajneet** [comment]
What is remaining before merge?

**Comment by Harjot1Singh** [comment]
Release to TestFlight + play store. Gonna switch to eas for building

File: src/with-contexts.tsx (hunk 1 above)
Comment (saihaj) [reply src/with-contexts.tsx:44]: I have ran issues on my vscode time to time where it slows down due to default exports, but that’s also in a monorpeo

**Pull request merged**